scripts/ci/travis/steps.sh

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bash
+# This script defines the test steps, used from run-step.sh.
+# In each step, the entire script is sourced first and subsequently one of the step functions is invoked.
+# Therefore, the global part of the script, outside the function defnitions, is run before every single steps.
+
+# TODO(niboshi): Definitions of the steps could be merged with scripts/ci/steps.sh.
+
+# Check preconditions
+test -f "$CHAINER_BASH_ENV"
+test -d "$REPO_DIR"
+test -d "$WORK_DIR"
+
+
+DEFAULT_JOBS=2
+
+
+_install_chainer_deps() {
+    # Install extras_require of Chainer.
+    local extras="$1"
+
+    # It's not possible to install only requirements.
+    # Chainer is uninstalled after the installation.
+    # TODO(niboshi): Use other installation tool
+    pip install -e "$REPO_DIR"["$extras"]
+    pip uninstall -y chainer
+}
+
+
+step_install_chainer_style_check_deps() {
+    _install_chainer_deps stylecheck
+}
+
+
+step_install_chainer_test_deps() {
+    _install_chainer_deps test
+
+    # Optional packages
+    local reqs=(
+        theano
+        h5py
+        pillow
+    )
+    pip install "${reqs[@]}"
+}
+
+
+step_before_install_chainer_test() {
+    # Remove oclint as it conflicts with GCC (indirect dependency of hdf5)
+    if [[ $TRAVIS_OS_NAME = "osx" ]]; then
+        brew update >/dev/null
+        brew outdated pyenv || brew upgrade --quiet pyenv
+
+        PYTHON_CONFIGURE_OPTS="--enable-unicode=ucs2" pyenv install -ks $PYTHON_VERSION
+        pyenv global $PYTHON_VERSION
+        python --version
+
+        brew install hdf5
+    fi
+}
+
+step_before_install_chainermn_test_deps() {
+    case "$TRAVIS_OS_NAME" in
+        linux)
+            local pkgs=(openmpi-bin openmpi-common libopenmpi-dev)
+            sudo apt-get install -y "${pkgs[@]}"
+            ;;
+        osx)
+            brew install open-mpi
+            ;;
+        *)
+            false
+            ;;
+    esac
+}
+
+
+step_install_chainermn_test_deps() {
+    pip install mpi4py
+}
+
+
+step_install_chainer_docs_deps() {
+    _install_chainer_deps docs
+}
+
+
+step_python_style_check() {
+    check_targets=(
+        "$REPO_DIR"/*.py
+        "$REPO_DIR"/chainer
+        "$REPO_DIR"/chainermn
+        "$REPO_DIR"/chainerx
+        "$REPO_DIR"/tests
+        "$REPO_DIR"/examples
+        "$REPO_DIR"/chainerx_cc/examples
+    )
+
+    # Check all targets exist
+    for f in "${check_targets[@]}"; do test -e "$f"; done
+
+    flake8 --version
+    flake8 "${check_targets[@]}"
+
+    false  # TODO: remove me
```

Comment:
This command emulates a test failure.
After review, I'll remove this line and confirm CI pass in turn.

Reply:
I think it's okay to merge, could you revert this temporary modification?